server/schedulers/balance_leader.go

```diff
@@ -249,20 +250,45 @@ func (l *balanceLeaderScheduler) IsScheduleAllowed(cluster schedule.Cluster) boo
 	return allowed
 }
 
+// candidateStores for balance_leader, order by `getStore` `asc``
 type candidateStores struct {
-	stores        []*core.StoreInfo
-	storeIndexMap map[uint64]int
-	index         int
-	compareOption func([]*core.StoreInfo) func(int, int) bool
+	stores   []*core.StoreInfo
+	getScore func(*core.StoreInfo) float64
```

Comment:
This function can be defined individually?

Reply:
Similar to the previous attribute `compareOption func([]*core.StoreInfo) func(int, int) bool`, making `getScore` as an attribute of `candidateStores` instead of an individually defined function would make the struct more independent and flexible. A simple example is that we could test it easily, do not need to care such as those policy, influence thing.